src/comms.cpp

```diff
@@ -859,6 +892,19 @@ void open_door() {
     door_command(DoorAction::Open);
 }
 
+void TTCdelayLoop() {
+    if (--TTCcountdown > 0) {
```

Comment:
> There is a "service_timers_loop" in the ratgdo.cpp file. Should this go there instead?

I probably erred in naming this xxxLoop as this function is not part of the main loop.  It executes once every 500ms using Ticker().  And it only executes during a time-to-close delay (so for a 10 second delay, it is called 20 times) to cycle the light and on completion of the timeout close the door.

Reply:
> Right now in the transmitSec2 function we write the rolling code to flash every time. I think we need to add something to only write once an hour or something. This should be ok now that the device is stable enough that it stays up for days at a time. The esphome variant does something along these lines.
> 
> I think we have to do this now because closing the garage door now will trigger a lot more writes to flash.

If we do this (which makes sense) then we need some sort of error recovery... as we now have EspSaveCrash with a callback then maybe we have to write to the flash whenever it crashes.  And is there a way to detect on boot-up that things are out-of-sync and re-initialize it... then I can remove the code file deletes from protocol change.

Reply:
for reference: https://github.com/ratgdo/esphome-ratgdo/blob/main/components/ratgdo/secplus2.cpp#L25

Reply:
Ah, so the GDO is tolerant of incrementing >1 in the rolling code, and ESPhome guys took a guess at 60.  We could do the same, but note that time-to-close light flash sends two per second, so our math might be a little different... if we write once a minute, then in theory we could have >120 code increments.  And we can also handle the crash case with our callback, and manually requested reboots.  Which just leaves power outages as uncaught.

Reply:
btw... what is the expected life of the ESP8266 flash?